android/app-example/src/main/java/com/badoo/ribs/example/rib/dialog_example/DialogExampleInteractor.kt

```diff
@@ -19,11 +21,11 @@ import com.jakewharton.rxrelay2.BehaviorRelay
 import io.reactivex.functions.Consumer
 
 class DialogExampleInteractor(
-    router: Router<Configuration, DialogExampleView>,
+    router: Router<Configuration, Nothing, Content, Overlay, DialogExampleView>,
```

Comment:
TBH, I feel like it is way too many type parameters here
Can we put them into configuration interface somehow?

Reply:
See next section why we can't. Feel free to suggest later if you have ideas how to keep compile-time safety, now going with current one.